src/backend/pluginLoaders.js

```diff
@@ -31,12 +32,24 @@ opaque type CachedProject = {|
   +project: Project,
 |};
 
+/**
+ * Represents all disjoint WeightedGraphs for a CachedProject.
+ */
+opaque type PluginGraphs = {|
+  +graphs: $ReadOnlyArray<WeightedGraphT>,
+  +cachedProject: CachedProject,
```

Comment:
why does PluginGraphs hold onto the cachedProject?

Reply:
More generally, I don't see why we have this type at all. Just return a read-only array of graphs.

Reply:
The type `$ReadOnlyArray<WeightedGraphT>` does not allow the assumption we can do:
`WeightedGraph.merge(graphs)` without any conflict resolution.
Or rather, that it's a plugin bug if it does cause an error to be thrown.

The opaque `PluginGraphs` does add this extra context, allowing the assumption.

The _only_ other context in which we can make this assumption is the `createPluginGraphs` function itself. So if we're ditching this type. I suggest we don't return `$ReadOnlyArray<WeightedGraphT>`, but a merged `WeightedGraphT`.

As to containing the `CachedProject`, it's similar to why `CachedProject` has the `Project`.
The following `contractIdentities` step doesn't make sense to apply to any _other_ Project than the one that created the `PluginGraphs`. Though removing the `CacheProvider` from this type would be an assumption about what the expected operations are.

Reply:
If we want to remove the `PluginGraphs` type I would suggest:

We should return a `WeightedGraphT`. Because only in this context do we know it's correct to let `WeightedGraph.merge(graphs)` crash if there's unresolved conflicts.

The new `contractPluginGraphs` signature then would be:
```js
// Before
(PluginLoaders, PluginGraphs) => WeightedGraphT
// After
(PluginLoaders, WeightedGraphT, Project) => WeightedGraphT
```

Moving the responsibility to *use* this function to offer identities to `LoadContext`, as well as to ensure the `Project` is the same as the one that produced the input `WeightedGraphT`.

Which is the opposite approach of what we settled on for `updateMirror` and `CachedProject`.

Reply:
Another alternative would be to make both `createPluginGraphs` and `contractPluginGraphs`  (#1619) private functions. So they're called by a public function which combines both. Same as
https://github.com/sourcecred/sourcecred/blob/806919a6ccd009d96b6645c2e8b6d6b68dd8ec91/src/api/loadWeightedGraph.js#L21-L41

Reply:
OK. I'm actually fine with either approach. Feels like an implementation detail to me so I'd rather go for velocity... approving.

Reply:
Since the current implementation is fairly local to the PluginLoaders and LoadContext. Changing it should be relatively easy. I'll go with the solution as-is, but would be happy to revisit in a follow-up if needed :]